docs/process/guidance/feature_flags/index.rst

```diff
@@ -0,0 +1,507 @@
+   with combinatorial configurations which can quickly explode the number of test
+   cases. Different levels of abstraction are also affected by different
+   restrictions, for unit testing it might be trivial to test all features but at
+   integration level it might be impossible to test all feature sets due to the
+   inherent complexity and resource limitations.
+
+   1. To avoid combinations, features **should** be independent of each other.
+      If there is a combination of settings that represent a feature, it might
+      make sense to bring them all together in a single flag.
+
+   2. To start off, we test two configurations:
+
+      - Configuration of the Core Software Stack
+      - Configuration for increased coverage of features
+
+      This restricts feature flags to at most two tested values.
+      It is recommended to not use more values for a feature flag.
+      On request, further configurations may be added after impact evaluation.
+
+   3. Features flags should be treated as inventory with a carrying cost.
+      In order to keep the number of feature flags manageable, it is recommended to
+      time-scope flags and teams must be proactive in its management. One
+      possible approach is to create an "expiration date" or "cleanup" task when
+      adding new flags. The responsible group for governance shall supervise this
+      practice but responsibility for maintenance must remain with the area.
+
+Feature flags structure
+-----------------------
+
+SCORE has four categories of feature flags.
+
+Generic requirements for feature flags:
+
+- Any feature flag shall be defined with a reasonable default value.
+- Features that are experimental must be clearly marked as such by prepending `experimental_`
+  (e.g. `experimental_<flag_name>`).
+- Above every feature flag in the BUILD file a comment must link to the relevant documentation.
+
+Below, we list the four categories of feature flags and provide additional information and constraints for each.
+
+Enabling a feature
+^^^^^^^^^^^^^^^^^^
+
+This feature flag is unique per feature in the feature tree and mentioned in the feature description at the top of the
+document.
+
+The name of the flag is defined as `<feature_name>`, where `<feature_name>` corresponds to the name of
+the feature in `snake_case`.
+
+The feature flag must be of type `bool_flag`.
+
+Feature flags of this category reside in
+`eclipse-score/score:flags`_.
+
+Selection of the implementation for an enabled feature
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A feature may be provided by multiple modules. The selection of the module which is used for a feature happens through
+a feature flag.
+
+The name of the flag is defined as `<feature_name>_implementation`, where `<feature_name>` corresponds to the name of
+the feature in `snake_case`.
+
+The feature flag must be of type `string_list_flag` and shall only allow single values.
```

Comment:
What are single values? I assume it's any of the modules that implement the feature? Is it the module name in snake_case?

Reply:
No it will depend on what modules make up the implementation of the feature. The feature documentation in the feature tree should give an explanation on what options there are and what they mean.